md-dao/src/main/java/com/wipro/ats/bdre/md/dao/JobDAO.java

```diff
@@ -258,24 +263,21 @@
                 batch.setBatchType("Type1");
                 session.save(batch);
 
-                targetBatchId = batch.getBatchId();
 
                 //update the batch consump queue table by assigning the target batch id to the max_batch no.of batches
                 // and updating the batch state
-                Criterion nullBCQ = Restrictions.isNull("batchCutPattern");
-                Criterion enptyBCQ = Restrictions.eq("batchCutPattern", "");
+                Criterion nullBCQ = Restrictions.isNull(BATCHCUTPATTERN);
+                Criterion enptyBCQ = Restrictions.eq(BATCHCUTPATTERN, "");
                 LogicalExpression orExp = Restrictions.or(nullBCQ, enptyBCQ);
 
-                Criteria blankBCPCriteria = session.createCriteria(Process.class).add(Restrictions.eq("process", parentProcess))
-                        .add(Restrictions.eq("deleteFlag", false));
+                Criteria blankBCPCriteria = session.createCriteria(Process.class).add(Restrictions.eq(PROCESS, parentProcess))
+                        .add(Restrictions.eq(DELETEFLAG, false));
                 blankBCPCriteria.add(orExp);
                 LOGGER.info("blankBCQCriteria size is" + blankBCPCriteria.list().size());
-                //Map<Integer,List<BatchConsumpQueue>> queueIdMap = new HashMap<Integer,List<BatchConsumpQueue>>();
                 for (Object blankBDPObject : blankBCPCriteria.list()) {
                     Process blankBCPProcess = (Process) blankBDPObject;
-                    Criteria entriesForBlankBCPCriteria = session.createCriteria(BatchConsumpQueue.class).add(Restrictions.eq("process", blankBCPProcess));
-                    // List<BatchConsumpQueue> BCQObjectList = new ArrayList<BatchConsumpQueue>();
-                    if (entriesForBlankBCPCriteria.list().size() > 0) {
+                    Criteria entriesForBlankBCPCriteria = session.createCriteria(BatchConsumpQueue.class).add(Restrictions.eq(PROCESS, blankBCPProcess));
+                    if (!entriesForBlankBCPCriteria.list().isEmpty()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)